model-optimizer/mo/main.py

```diff
@@ -97,16 +97,25 @@ def print_argv(argv: argparse.Namespace, is_caffe: bool, is_tf: bool, is_mxnet:
 
 
 def prepare_ir(argv: argparse.Namespace):
-    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx = deduce_framework_by_namespace(argv)
-
     fem = argv.feManager
-    new_front_ends = []
-    if fem is not None:  # in future, check of 'use_legacy_frontend' in argv can be added here
-        new_front_ends = fem.get_available_front_ends()
+    available_moc_front_ends = []
+    moc_front_end = None
+    if fem:  # in future, check of 'use_legacy_frontend' in argv can be added here
+        available_moc_front_ends = fem.get_available_front_ends()
+        if argv.input_model:
+            if not argv.framework:
+                moc_front_end = fem.load_by_model(argv.input_model)
+                if moc_front_end:
+                    argv.framework = moc_front_end.get_name()
+            elif argv.framework in available_moc_front_ends:
+                moc_front_end = fem.load_by_framework(argv.framework)
+
+    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx =\
+        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False]
 
     if not any([is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx]):
```

Comment:
there is no variable for `is_paddle`. we can handle errors related to incorrectly specified model here.

Reply:
Yes, it is already done. Inside this 'if' we're checking if 'model' is supported by any of MOC frontends (not only Paddle).
See `if argv.framework not in frameworks:`